gcloud-java-examples/src/main/java/com/google/cloud/examples/nio/ParallelCountBytes.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.examples.nio;
+
+import com.google.common.base.Stopwatch;
+
+import javax.xml.bind.annotation.adapters.HexBinaryAdapter;
```

Comment:
I would prefer not to use this (it's very out of context here and I believe not available in Android).

Instead we could use Guava's [Hasher](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/hash/Hasher.html). The code becomes:
```java
Hasher hasher = Hashing.md5().newHasher();
for (int block = 0; block < nBlocks; block++) {
  BufWithLock bwl = bufs[block % bufs.length];
  synchronized (bwl.lock) {
    while (!bwl.full) {
      bwl.lock.wait();
    }
    hasher.putBytes(bwl.buf.array(), 0, bwl.buf.position());
    total += bwl.buf.position();
    bwl.full = false;
    bwl.lock.notifyAll();
  }
}
long elapsed = sw.elapsed(TimeUnit.SECONDS);
System.out.println("Read all " + total + " bytes in " + elapsed + "s. ");
System.out.println("The MD5 is: 0x" + hasher.hash().toString());
```